client/src/pages/private/ParticipantView.js

```diff
@@ -584,6 +594,45 @@ export default () => {
           }}
           onClose={defaultOnClose}
         />}
+        {activeModalForm === 'edit-participant' && <EditParticipantForm
+          initialValues={{
+            ...actionMenuParticipant
+          }}
+          validationSchema={EditParticipantFormSchema}
+          onSubmit={async (values) => {
+            const history = {
+              timestamp: new Date(),
+              changes: [],
+            };
+            Object.keys(values).forEach(key => {
+              if (values[key] !== actionMenuParticipant[key]) {
+                history.changes.push({
+                  field: key,
+                  from: actionMenuParticipant[key],
+                  to: values[key],
+                });
+              }
+            });
+            values.history = (actionMenuParticipant.history)? [history, ...actionMenuParticipant.history] : [history];
```

Comment:
Should these be in chronological order?
```suggestion
            values.history = (actionMenuParticipant.history)? [...actionMenuParticipant.history, history] : [history];
```